hub/dataload/data_parsers.py

```diff
@@ -1,35 +1,124 @@
-import networkx
-import obonet
 import os
+import re
+from collections import defaultdict
+
+import networkx as nx
+import obonet
+
+
+class OntologyHelper:
+    IS_A_EDGE_TYPE = "is_a"
+    SYNONYM_PATTERN = re.compile(r"\"(.+?)\"")
+    XREF_INVALID_PREFIXES = {"https", "http"}
+    XREF_ALWAYS_PREFIXED = {"DOID", "HP", "MP", "OBI", "EFO"}
+
+    def __init__(self, prefix):
+        self.prefix = prefix
+
+    def load_obo_network(self, filepath: str) -> nx.MultiDiGraph:
+        graph = obonet.read_obo(filepath, ignore_obsolete=False)
+        edges_to_remove = [(u, v, key) for (u, v, key) in graph.edges(keys=True) if key != self.IS_A_EDGE_TYPE]
+        graph.remove_edges_from(edges_to_remove)
+        return graph
+
+    def parse_synonyms(self, node_obj: dict) -> dict:
+        if "synonym" not in node_obj:
+            return {}
+        exact_synonyms = []
+        related_synonyms = []
+        for synonym_description in node_obj["synonym"]:
+            if "EXACT" in synonym_description:
+                exact_synonyms += self.SYNONYM_PATTERN.findall(synonym_description)
+            elif "RELATED" in synonym_description:
+                related_synonyms += self.SYNONYM_PATTERN.findall(synonym_description)
+        synonyms = {}
+        if exact_synonyms:
+            synonyms["exact"] = exact_synonyms
+        if related_synonyms:
+            synonyms["related"] = related_synonyms
+        return synonyms
+
+    def parse_xref(self, node_obj: dict) -> dict:
+        if "xref" not in node_obj:
+            return {}
+        xrefs = defaultdict(set)
+        for curie in node_obj.get("xref"):
+            curie_prefix, curie_id = curie.split(":", 1)
+            if curie_prefix in self.XREF_INVALID_PREFIXES:
```

Comment:
Do we receive queries that use http and https as prefixes such that we need to check the CURIE prefix against it?